osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/tasks/ScheduledTaskExecutorService.java

```diff
@@ -111,26 +126,43 @@ private void processScheduledTasks(final ScheduledTaskStatusType type) {
     List<ScheduledTask> scheduledTasks = this.getScheduledTasks(type);
 
     while (!scheduledTasks.isEmpty()) {
-      for (ScheduledTask scheduledTask : scheduledTasks) {
-        LOGGER.info(
-            "Processing scheduled task for device [{}] to perform [{}]  ",
-            scheduledTask.getDeviceIdentification(),
-            scheduledTask.getMessageType());
-        try {
-          scheduledTask.setPending();
-          scheduledTask = this.scheduledTaskRepository.save(scheduledTask);
-          final ProtocolRequestMessage protocolRequestMessage =
-              this.createProtocolRequestMessage(scheduledTask);
-          this.deviceRequestMessageService.processMessage(protocolRequestMessage);
-        } catch (final FunctionalException e) {
-          LOGGER.error("Processing scheduled task failed.", e);
-          this.scheduledTaskRepository.delete(scheduledTask);
+      final ExecutorService executorService =
+          Executors.newFixedThreadPool(
+              this.scheduledTaskExecutorJobConfig.getScheduledTaskThreadPoolSize());
+      try {
+        final List<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (final ScheduledTask scheduledTask : scheduledTasks) {
+          final CompletableFuture<Void> future =
+              CompletableFuture.runAsync(
+                  () -> this.processScheduledTask(scheduledTask), executorService);
+          futures.add(future);
         }
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+      } finally {
+        shutdownAndAwaitTermination(executorService, Duration.ofSeconds(AWAIT_TERMINATION_IN_SEC));
```

Comment:
google functionaliteit, ben ik wel voor. Deze is wel @Beta, dus zou kunnen veranderen/verdwijnen.

Reply:
Heb hem vervangen door eigen implementatie

Reply:
Heb het weer terug gedraaid na overleg